src/client.ts

```diff
@@ -4627,22 +4627,24 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             threadId = null;
         }
         const reason = opts?.reason;
+        let withRelTypes = opts?.with_rel_types ?? opts?.with_relations;
 
-        if (
-            opts?.with_relations &&
-            this.canSupport.get(Feature.RelationBasedRedactions) === ServerSupport.Unsupported
-        ) {
+        if (typeof withRelTypes === "string") {
+            withRelTypes = [withRelTypes];
+        }
```

Comment:
why is this required? `IRedactOpts` does not permit a string here?

Reply:
The previous version allowed string or list. But since it is an unstable function anyway, I will remove the fallback.

Reply:
> The previous version allowed string or list.

Are you sure? https://github.com/matrix-org/matrix-js-sdk/blob/e10db6f7e9978e2d8dd3640fbe2be503710fd506/src/%40types/requests.ts#L61 requires it to be an array.

Reply:
You are right. Also the MSC didn't allow it to be a string. Anyway removing this block then makes event more sense.